extensions/sql-migration/src/wizard/skuRecommendation/skuRecommendationPage.ts

```diff
@@ -871,446 +340,151 @@ export class SKURecommendationPage extends MigrationWizardPage {
-					await this._skuRestartDataCollectionButton.updateCssStyles({ 'display': 'block' });
-					await this._refreshAzureRecommendationButton.updateCssStyles({ 'display': 'none' });
-					await this._skuEditParametersContainer.updateCssStyles({ 'display': 'block' });
-				}
+		// dbReady are thos databases without issues. dbNotReady are those databases with atleast one issue with issueCategory = "Issue".
+		// dbReadyWithWarnings can be found by subtraction dbReady and dbNotReady with total dbCount.
+		const dbReady = this.migrationStateModel._assessmentResults?.databaseAssessments?.filter(db =>
+			!db.issues?.some(issue => issue.appliesToMigrationTargetPlatform === targetType)
+		).length;
+		const dbNotReady = this.migrationStateModel._assessmentResults?.databaseAssessments?.filter(db =>
+			db.issues?.some(issue => (issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === IssueCategory.Issue))
+		).length;
+		const dbReadyWithWarnings = dbCount - (dbReady + dbNotReady);
+
+		// blockers contain count of all the instance level issues with issueCategory = "Issue".
+		var blockers = this.migrationStateModel._assessmentResults?.issues.filter(issue =>
+			(issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === IssueCategory.Issue)).length;
+		// also blockers includes sum of all the dabatase level issues with issueCategory = "Issue" for each database.
+		blockers += this.migrationStateModel._assessmentResults?.databaseAssessments?.reduce((count, database) =>
+			count + database.issues.filter(issue => (issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === IssueCategory.Issue)).length, 0);
+
+		// warnings contain count all the instance level issues with issueCategory = "Warning".
+		var warnings = this.migrationStateModel._assessmentResults?.issues.filter(issue =>
+			(issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === IssueCategory.Warning)).length;
+		// also warnings includes sum of all the dabatase level issues with issueCategory = "Warning" for each database.
+		warnings += this.migrationStateModel._assessmentResults?.databaseAssessments?.reduce((count, database) =>
+			count + database.issues.filter(issue => (issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === IssueCategory.Warning)).length, 0);
+
+		switch (targetType) {
+			case MigrationTargetType.SQLDB:
+				this._dbAssessmentCard.updateAssessmentResult(dbCount, dbReady, dbReadyWithWarnings, dbNotReady, blockers, warnings);
 				break;
-			}
-
-			case PerformanceDataSourceOptions.OpenExisting: {
-				await this._azureRecommendationSectionText.updateProperties({
-					description: constants.AZURE_RECOMMENDATION_TOOLTIP_NOT_STARTED
-				});
+			case MigrationTargetType.SQLMI:
+				this._miAssessmentCard.updateAssessmentResult(dbCount, dbReady, dbReadyWithWarnings, dbNotReady, blockers, warnings);
+				break;
```

Comment:
For both these targets action seams similar, then why we put them in switch-case?